packages/invoice-dashboard/src/lib/dashboard/invoice-view.svelte

```diff
@@ -469,6 +479,58 @@
       : value;
   }
 
+  function formatBalance(value: number, maxDecimals: number = 4): string {
+    return Number.isInteger(value)
+      ? value.toString()
+      : value.toFixed(maxDecimals);
+  }
+
+  async function checkBalance() {
+    try {
+      if (!address || !paymentCurrencies[0] || !network) {
+        console.log("Missing required parameters for balance check:", {
+          address,
+          paymentCurrency: paymentCurrencies[0],
+          network,
+        });
+        return;
+      }
+
+      const invoiceNetworkId = Number(getNetworkIdFromNetworkName(network));
+
+      if (account.chainId !== invoiceNetworkId) {
+        hasEnoughBalance = false;
+        console.log("Wrong network - balance check skipped");
+        return;
+      }
+
+      if (paymentCurrencies[0]?.type === Types.RequestLogic.CURRENCY.ERC20) {
```

Comment:
I created an issue to fix our usage of `paymentCurrencies[0]`
* https://github.com/RequestNetwork/web-components/issues/260

Reply:
Sure, but this is out of scope for this task

Reply:
@MantisClone Will take a look at this one later

Reply:
Yes exactly. This is out of scope so I made a separate issue to fix it later.